MdeModulePkg/Core/Dxe/Mem/Page.c

```diff
@@ -2028,6 +2200,30 @@ CoreGetMemoryMap (
             (Entry->End   <= mMemoryTypeStatistics[Type].MaximumAddress))
         {
           MemoryMap->Type = Type;
+        } else if (mMemoryTypeStatistics[Type].Special &&
+                   (mMemoryTypeStatistics[Type].NumberOfPages > 0) &&
+                   MemoryRegionsIntersect (
+                     Entry->Start,
+                     Entry->End,
+                     mMemoryTypeStatistics[Type].BaseAddress,
+                     mMemoryTypeStatistics[Type].MaximumAddress
+                     ))
+        {
+          // There is partial overlap with a special memory type bin.

```

Comment:
How would this condition occur of the rest of the logic in this PR is correct?  Would this only occur of the internal memory map structures are corrupt?

Reply:
This check is in place to ensure the returned memory is sane, because this is what the non-core consumers will be consuming.

This overlap should not occur, if all memory map manipulation goes through the `CoreAddRange` function. But I did notice that whenever the memory map is being "converted" during allocation or free, that routine does not go through this same function.